File: duplicatesFromSortedArray2.cpp
// Time Complexity :
// Space Complexity :
// Did this code successfully run on Leetcode :
// Three line explanation of solution in plain english

// Your code here along with comments explaining your approach

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size()<3) //if the size of the array is less than the 3 then return the size of the array
            return nums.size();
        
        for(int i=0;i!=nums.size()-2;i++){
           // cout<<i<<" "<<endl;
            if (nums[i]==nums[i+1] && nums[i]==nums[i+2]) {
                nums.erase(nums.begin()+i); //erase the first number in the group of numbers
                i--;
            }
        }
        return nums.size(); //return the size
    }
};

File: duplicatesfromsortedarray2.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length==0 || nums==null)    return 0;
        int count=1; int j=1; int k=2;
        for(int i=1;i<nums.length;i++)
        {
            if(nums[i]==nums[i-1])
            {
                count++;
            }
            else
            {
                count=1;
            }
            if(k>=count)
            {
                nums[j]=nums[i];
                j++;
            }
        }
        return j;
    }
}
//time complexity : O(n)
//space complexity: O(1)